cluster/gce/util.sh

```diff
@@ -754,6 +754,19 @@ function construct-linux-kubelet-flags {
   # Keep in sync with the mkdir command in configure-helper.sh (until the TODO is resolved)
   flags+=" --cert-dir=/var/lib/kubelet/pki/"
 
+  # Enabling KubeletCredentialProviders and DisableKubeletCloudCredentialProviders
+  # enables kubelet to use out-of-tree auth credential provider instead of in-tree 
+  # auth credential provider.
+  # https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider
+  if [[ "${ENABLE_AUTH_PROVIDER_GCP:-false}" == "true" ]]; then
+    flags+="  --feature-gates=DisableKubeletCloudCredentialProviders=true,KubeletCredentialProviders=true"
```

Comment:
This seems a bit fragile, because AFIAU if someone passes `KUBE_FEATURE_GATES`, kube-up will already pass `--feature-gates` : https://github.com/kubernetes/kubernetes/blob/5d446b205e308e54119320b5645fe17ce88d39cf/cluster/gce/gci/configure-helper.sh#L1737

As a result, if someone passes both `KUBE_FEATURE_GATES=some-gates...` and `ENABLE_AUTH_PROVIDER_GCP=true`, I think `--feature-gates` would be passed twice?

I would recommend to use the existing `KUBE_FEATURE_GATES` flag to specify `DisableKubeletCloudCredentialProviders=true,KubeletCredentialProviders=true` and to have `ENABLE_AUTH_PROVIDER_GCP` only configure the `--image-credential-provider-config=` and `--image-credential-provider-bin-dir=` flags.